src/hzb_combinatorial_libraries/schema.py

```diff
@@ -752,4 +783,92 @@ def normalize(self, archive, logger: BoundLogger) -> None:
         super(UnoldThermalEvaporation, self).normalize(archive, logger)
 
 
+class PixelProperty(EntryData):
+    m_def = Section(
+        categories=[UnoldLabCategory],
+        label='UnoldPixelProperty'
+    )
+    conductivity = Quantity(
+        type=np.dtype(np.float64),
+        # unit='1/Ω',
+        shape=[],
+        description="""
+                The conductivity of the Pixel.
+                """,
+        a_eln=dict(
+            component='NumberEditQuantity')
+    )
+    bandgap = Quantity(
+        type=np.dtype(np.float64),
+        description='Band gap value in eV.',
+        a_eln=dict(component='NumberEditQuantity')
+    )
+
+    PLQY = Quantity(
+        type=np.dtype(np.float64),
+        # unit='%',
+        description='Energy integrated value of the PL spectrum.',
+        a_eln=dict(component='NumberEditQuantity')
+    )
+
+    implied_voc = Quantity(
+        type=np.dtype(np.float64),
+        # unit='V',
+        description='Estimated open circuit voltage based on PL measurements.',
+        a_eln=dict(component='NumberEditQuantity')
+    )
+
+    FWHM = Quantity(
+        type=np.dtype(np.float64),
+        # unit='eV', # todo check if this is correct
+        description='FWHM based on PL measurements.',
+        a_eln=dict(component='NumberEditQuantity')
+    )
+
+
+class Pixel(ContinuousCombiSample, EntryData, ArchiveSection):
+    m_def = Section(
+        categories=[UnoldLabCategory],
+        label='UnoldPixel'
+    )
+    thickness = Quantity(
+        type=np.dtype(np.float64),
+        unit='nm',
+        shape=[],
+        description="""
+            The thickness of the pixel.
+            """,
+        a_eln=dict(
+            component='NumberEditQuantity')
+    )
+    properties = SubSection(
+        section_def=PixelProperty,
+        description='''
+          The properties of the pixel.
+          ''',
+    )
+    samples = SubSection(
+        section_def=CompositeSystemReference,
+        description='''
+          The samples as that have undergone the process.
+          ''',
+        repeats=True,
+    )
+
+    def normalize(self, archive, logger):
+        super(ContinuousCombiSample, self).normalize(archive, logger)
+
+        # self.components for xrf, check htem how to do it, and add element to results.materials.elements
+        id = self.lab_id.split(':')[0].strip()
```

Comment:
this should be after the following if statement will raise an error if no id is present else

Reply:
here should raise err or just leave it there ? this err will appear in the normalizing processing and reprocessing, right?

Reply:
yes and this error will appear when you create a pixel by hand